daemon/indexing/apollo/index.js

```diff
@@ -4,37 +4,170 @@ var search = require('../lib/search.js')
 var db = require('../lib/db.js')
 
 /*
- * Implentation of a GraphQL server using the Apollo framework.
- * See https://www.apollographql.com/server
+ * Implementation of the Origin GraphQL server.
+  * Uses the Apollo framework: https://www.apollographql.com/server
  *
  */
 
 // Type definitions define the "shape" of the data and specify
 // which ways the data can be fetched from the GraphQL server.
 const typeDefs = gql`
+  ######################
+  #
+  # Query output schema.
+  #
+  ######################
+
+  type User {
+    walletAddr: ID!   # Ethereum wallet address
+    identityAddr: ID  # ERC 725 identity address.
+  }
+
+  type Offer {
+    id: ID!
+    ipfsHash: ID!
+    listingId: ID!
+    buyer: User!
+    # Per contract definition: 1: Created, 2: Accepted, 3: Disputed
+    # NOTE: There is no "Finalized" status stored on-chain but could be useful
+    #       to compute it when resolving this object.
+    status: Int!
+  }
+
+  type Price {
+    currency: String!
+    amount: Float!
```

Comment:
Prices should definitely not be floats. Javascript doesn't have enough precision to handle ETH.

We might have to have this be a string. (Which can then be parsed client-side by the big number library.)

Reply:
I actually think ETH would be ok. AFAIK, Javascript uses 64 bits floats encoded with IEEE-754 which should give us something like 15 digits. So I can't really think of any item price in ETH that would cause precision loss ? But who knows, in the future we may support currencies with high inflation rate like zimbabwean dollar ! ;) Or possibly this API may get used by a language or library that deserializes JSON float using 32 bits. So overall better to use string.

And yes, also agreed that for doing math on those prices origin-js and the dapp should use a proper library like big-number or maybe currency.js

Reply:
One ETH is 10^18 wei, and the contracts all operate on wei. This means that origin-js needs to be accurate-to-the-wei, and with it the search server.

Reply:
Ah yep, makes sense. Changed to String.